pyuvsim/uvsim.py

```diff
@@ -280,12 +280,11 @@ def uvdata_to_task_iter(task_ids, input_uv, catalog, beam_list, beam_dict, Nsky_
         antennas.append(Antenna(antname, num, antpos_enu[num], beam_id))
 
     baselines = {}
-    Ntimes = input_uv.Ntimes
     Nfreqs = input_uv.Nfreqs
-    Nbls = input_uv.Nbls
+    Nblts = input_uv.Nblts
 
-    tasks_shape = (Ntimes, Nfreqs, Nbls)
-    time_ax, freq_ax, bl_ax = range(3)
+    tasks_shape = (Nblts, Nfreqs)
```

Comment:
I think we can get that same kind of behavior back, but it would require a re-evaluation of the make_task_inds and things like that. Basically it would need to iterate over unique times, then for each time find baseline which have that time, then iterate through each frequency. A little more logic mostly, just can't use the easy `unravel_index` from numpy.

Reply:
The difficulty here seems to be just in designing for the "non-square" case, when some times are missing baselines. I'm not sure if it would be faster to just drop computed visibilities that aren't requested, or to find some way to skip computing them.

I'm glad to see that you still require the `blt` ordering to be times then baselines. At no point should the final order of data in the file dictate the order of calculations within the task loop.

Reply:
I've been thinking about this, we always know that nblts * nfreqs arrays will be "square". what if we did something like this when task ids. It would require a memory overhead to generate the order array :thinking:  but it would force the ids to be iterated over in the time, baseline, freq order.
```
        _times = np.tile(input_uv.time_array, (input_uv.Nfreqs, 1)).T.flatten()[task_ids]
        _bls = np.tile(input_uv.baseline_array, (input_uv.Nfreqs, 1)).T.flatten()[task_ids]
        _freqs = np.tile(input_uv.freq_array.flatten(), (input_uv.Nblts, 1)).flatten()[task_ids]
        order = np.lexsort((_freqs, _bls, _times))

        # for task_index in task_ids:
        for index in order:
            task_index = task_ids[index]
```